### PR TITLE
[SM-943] [BEEEP] Swap to SQLite in-memory for integration tests

### PR DESCRIPTION
**Pull request by Thomas-Avery**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The purpose of this PR is to swap from using the EF Core in-memory database provider to the SQLite in-memory database provided for integration tests. 

The EF Core in-memory database provider has limitations and will not be able to support future SM updates:

- Does not support transactions
- Does not support `ExecuteUpdate` and `ExecuteDelete`
- and [more](https://learn.microsoft.com/en-us/ef/core/testing/choosing-a-testing-strategy#in-memory-as-a-database-fake)

The recommendation is to use the SQLite in-memory provider instead. 

## Code changes

* **bitwarden_license/test/Scim.IntegrationTest/Controllers/v2/GroupsControllerTests.cs:**
**bitwarden_license/test/Scim.IntegrationTest/Controllers/v2/UsersControllerTests.cs:**

Previously, we were using database names to provide a mechanism for sharing database between different factories. 
With Sqlite each `_connection.Open();` creates its own database. 
To share a database instance, the connection can be passed into multiple factories. 

For these Scim controller tests, each was providing a different database name, so we should be good to just use the default, thus creating a new DB for each.

* **test/Api.IntegrationTest/Factories/ApiApplicationFactory.cs:**
For API integration tests, we want to share a database between the `ApiApplicationFactory` and the `IdentityApplicationFactory`. 
Adding opening a connection in the `ApiApplicationFactory` and passing it into `SqliteConnection` for both factories.

* **test/Api.IntegrationTest/SecretsManager/Controllers/AccessPoliciesControllerTests.cs:**
**test/Api.IntegrationTest/SecretsManager/Controllers/SecretsControllerTests.cs:**
Since SQLite enforces foreign key constraints, updating tests that require entity initialization.

* **test/Api.IntegrationTest/SecretsManager/Controllers/ServiceAccountsControllerTests.cs:**
Fixing bug discovered by foreign key constraints.

* **test/Api.IntegrationTest/SecretsManager/SecretsManagerOrganizationHelper.cs:**
Adding a helper for creating a new SM organization.

* **test/IntegrationTestCommon/Factories/WebApplicationFactoryBase.cs:**
Swapping from EF Core in-memory to SQLite in-memory.
Exposing `SqliteConnection` to allow for sharing the same db between multiple factories.
Adding code to run EF migrations since SQLite requires running EF migrations on the in-memory database during setup.

* **bitwarden_license/test/Scim.IntegrationTest/packages.lock.json:**
**test/Api.IntegrationTest/packages.lock.json:**
**test/Identity.IntegrationTest/packages.lock.json:**
**test/IntegrationTestCommon/IntegrationTestCommon.csproj:**
**test/IntegrationTestCommon/packages.lock.json:**
Removing Microsoft.EntityFrameworkCore.InMemory dependency.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by cd-bitwarden — APPROVED** [review]
LGTM, great work Thomas 😄

**Review by justindbaur — APPROVED** [review]
Looks great, thanks!

File: test/Api.IntegrationTest/Factories/ApiApplicationFactory.cs [hunk 1]
```diff
@@ -2,17 +2,24 @@
...
 public class ApiApplicationFactory : WebApplicationFactoryBase<Startup>
 {
     private readonly IdentityApplicationFactory _identityApplicationFactory;
+    private readonly SqliteConnection _connection;
+    private const string _connectionString = "DataSource=:memory:";
 
     public ApiApplicationFactory()
     {
+        _connection = new SqliteConnection(_connectionString);
+        _connection.Open();
```
Comment (Hinton) [inline test/Api.IntegrationTest/Factories/ApiApplicationFactory.cs:18]: **nitpick (non-blocking)**: You could skip the whole `_connection` member and assign directly to `SqliteConnection`.

```c#
SqliteConnection = new SqliteConnection(_connectionString);
SqliteConnection.Open();

_identityApplicationFactory = new IdentityApplicationFactory();
_identityApplicationFactory.SqliteConnection = SqliteConnection;
```

**Review by Hinton — DISMISSED** [review]
One non-blocking comment.

File: test/Api.IntegrationTest/Factories/ApiApplicationFactory.cs (hunk 1 above)
Comment (Thomas-Avery) [reply test/Api.IntegrationTest/Factories/ApiApplicationFactory.cs]: Seems worth making the change now vs later. 

https://github.com/bitwarden/server/pull/3292/commits/6048b774dd571118b1d85a3f5e156eb9928cbc23

**Review by Hinton — APPROVED** [review]

**Review by justindbaur — APPROVED** [review]

**Review by cd-bitwarden — APPROVED** [review]
LGTM!

**Pull request merged**